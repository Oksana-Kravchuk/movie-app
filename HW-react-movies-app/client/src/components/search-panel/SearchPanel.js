import React from 'react';
import PropTypes from 'prop-types';
import { Input, Button } from 'antd';

import './SearchPanel.scss';

// PATTERN: Stateless Component

const SearchPanel = ({ handleSearch, handleInput }) => {
  return (
    <div className="search-panel">
      <h2 className="search-panel__header">Find your movie</h2>
      <div className="search-panel__row">
        <Input
          placeholder="What do you want to watch?"
          size="large"
          onChange={(e) => handleInput(e.target.value)}
          onKeyPress={(e) => e.key === 'Enter' && handleSearch()}
        />
        <Button
          type="primary"
          size="large"
          className="search-panel__button"
          onClick={handleSearch}
        >
          Search
        </Button>
      </div>
    </div>
  );
};

SearchPanel.propTypes = {
  handleSearch: PropTypes.func,
  handleInput: PropTypes.func,
};

export default SearchPanel;
