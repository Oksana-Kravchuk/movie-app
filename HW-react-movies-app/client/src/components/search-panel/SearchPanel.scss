@import '../../common/styles/colours';

.search-panel {
  margin: 80px 20% 0;

  &__header {
    font-size: 30px;
    text-transform: uppercase;
    font-weight: 200;
    margin-bottom: 30px;
    letter-spacing: 2.4px;
    color: $white;
  }

  input {
    border: 0;
    background: $secondary;
    height: 50px;
    opacity: 0.8;
    border-radius: 2px;
    color: $white;
    margin-right: 10px;

    &::placeholder {
      color: $grey;
      font-size: 18px;
    }
  }

  &__row {
    display: flex;
  }

  &__button {
    height: 50px;
    text-transform: uppercase;
    width: 200px;
    letter-spacing: 1.5px;
  }
}
