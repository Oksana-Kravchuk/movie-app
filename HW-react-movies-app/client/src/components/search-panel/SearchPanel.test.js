import React from 'react';
import { Button } from 'antd';
import { shallow } from 'enzyme';
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from 'enzyme-to-json';
import SearchPanel from './SearchPanel';

Enzyme.configure({ adapter: new Adapter() });

describe('AddMovieContainer ', () => {
  let wrapper;
  let handleSearch;
  let handleInput;

  beforeEach(() => {
    handleSearch = jest.fn();
    handleInput = jest.fn();
    const props = {
      handleSearch,
      handleInput
    };
    wrapper = shallow(<SearchPanel {...props} />);
  });

  it('render correctly SearchPanel component', () => {  
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should run handleInput function on input', () => {  
    const input = wrapper.find('Input');

    input.simulate('change', { target: {value: 'a' } });
    expect(handleInput).toHaveBeenCalledWith('a');
  });

  it('should run handleSearch function on buttton click', () => {  
    const button = wrapper.find(Button);

    button.simulate('click');
    expect(handleSearch).toHaveBeenCalled();
  });
});