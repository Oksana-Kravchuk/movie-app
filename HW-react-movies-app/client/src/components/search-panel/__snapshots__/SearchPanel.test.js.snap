// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieContainer  render correctly SearchPanel component 1`] = `
<div
  className="search-panel"
>
  <h2
    className="search-panel__header"
  >
    Find your movie
  </h2>
  <div
    className="search-panel__row"
  >
    <Input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="What do you want to watch?"
      size="large"
      type="text"
    />
    <Button
      block={false}
      className="search-panel__button"
      ghost={false}
      htmlType="button"
      loading={false}
      onClick={[MockFunction]}
      size="large"
      type="primary"
    >
      Search
    </Button>
  </div>
</div>
`;
